51-android.rules

```diff
@@ -797,91 +797,130 @@ LABEL="not_XiaoMi"
-ATTR{idProduct}=="1485", ENV{adb_user}="yes"
+ATTR{idProduct}=="1485", GOTO="user"
 #   MF286R internal LTE modem
-ATTR{idProduct}=="1489", ENV{adb_user}="yes"
+ATTR{idProduct}=="1489", GOTO="user"
 #   Nubia / RedMagic Series (NX***)
 #   See https://github.com/TadiT7/nubia_nx619j_dump/blob/NX619J-user-9-PKQ1.180929.001-eng.nubia.20181220.181559-release-keys/vendor/etc/init/hw/init.nubia.usb.rc
 #   ptp,adb
-ATTR{idProduct}=="ffd1", ENV{adb_adb}="yes"
+ATTR{idProduct}=="ffd1", GOTO="adb"
 #   mtp,adb
-ATTR{idProduct}=="ffcf", ENV{adb_adb}="yes"
+ATTR{idProduct}=="ffcf", GOTO="adb"
 #   mass_storage,adb
-ATTR{idProduct}=="ffcd", ENV{adb_adb}="yes"
+ATTR{idProduct}=="ffcd", GOTO="adb"
 #   rndis,adb
-ATTR{idProduct}=="ffcb", ENV{adb_adb}="yes"
+ATTR{idProduct}=="ffcb", GOTO="adb"
 #   modem,service,nema,adb
-ATTR{idProduct}=="ffc9", ENV{adb_adb}="yes"
+ATTR{idProduct}=="ffc9", GOTO="adb"
 #   modem,service,nema,mass_storage,adb
-ATTR{idProduct}=="ffc7", ENV{adb_adb}="yes"
+ATTR{idProduct}=="ffc7", GOTO="adb"
 #   diag,modem,mass_storage,adb
-ATTR{idProduct}=="ffb0", ENV{adb_adb}="yes"
+ATTR{idProduct}=="ffb0", GOTO="adb"
 #   diag,modem,service,mass_storage,adb
-ATTR{idProduct}=="ffb2", ENV{adb_adb}="yes"
+ATTR{idProduct}=="ffb2", GOTO="adb"
 #   adb
-ATTR{idProduct}=="ffc1", ENV{adb_adb}="yes"
+ATTR{idProduct}=="ffc1", GOTO="adb"
 #   diag,mass_storage,adb
-ATTR{idProduct}=="ffc0", ENV{adb_adb}="yes"
-GOTO="android_usb_rule_match"
+ATTR{idProduct}=="ffc0", GOTO="adb"
+GOTO="android_usb_rules_end"
 LABEL="not_ZTE"
 
 # ZUK
-ATTR{idVendor}=="2b4c", ENV{adb_user}="yes"
+ATTR{idVendor}=="2b4c", GOTO="user"
 
 # Verifone
-ATTR{idVendor}=="11ca", ENV{adb_user}="yes"
+ATTR{idVendor}=="11ca", GOTO="user"
+
+# No android devices found
+GOTO="android_usb_rules_end"
+
+# ADB Debug User (default)
+LABEL="adb", ENV{adb_adb}="yes", GOTO="adb_set_values"
+
+# ADB Debug and Audio Source
+LABEL="adbaud", ENV{adb_adb}="yes"
+LABEL="aud", ENV{adb_user}="yes", GOTO="adb_set_values"
+
+# ADB Debug and AT-commands CDC Serial
+LABEL="adbcdc", ENV{adb_adb}="yes"
+LABEL="cdc", ENV{adb_user}="yes", GOTO="adb_set_values"
+
+# ADB Debug and Fastboot mode
+LABEL="adbfast", ENV{adb_adbfast}="yes", GOTO="adb_set_values"
+
+# ADB Debug and mass storage (note: android_ver<4.3)
+LABEL="adbmass", ENV{adb_adbmass}="yes"
+LABEL="mass", ENV{adb_mass}="yes", GOTO="adb_set_values"
+
+# ADB Debug and MIDI mode (also check to see /dev/midi%n)
+LABEL="adbmidi", ENV{adb_adbmidi}="yes"
+LABEL="midi", ENV{adb_user}="yes", GOTO="adb_set_values"
+
+# ADB Debug and MTP mode
+LABEL="adbmtp", ENV{adb_adbmtp}="yes"
+LABEL="mtp", ENV{adb_mtp}="yes", GOTO="adb_set_values"
+
+# ADB Debug and PTP mode
+LABEL="adbptp", ENV{adb_adbptp}="yes"
+LABEL="ptp", ENV{adb_ptp}="yes", GOTO="adb_set_values"
+
+# ADB Debug and Tether mode
+LABEL="adbrndis", ENV{adb_adb}="yes"
+LABEL="rndis", ENV{adb_user}="yes", GOTO="adb_set_values"
 
-# Skip other vendor tests
-LABEL="android_usb_rule_match"
```

Comment:
Good catch, I thought of reducing the label size to "adb_set_values" , while doing the labels above, and missed those. It was part of the reducing the "go_....." to smaller labels.

Question: Change the remaining "android_usb_rule_match" to "adb_set_values" or change the existing "adb_set_values" to "android_usb_rule_match"?

Reply:
`adb_set_values` feels a bit more descriptive, but now that I looked at it more closely I don't think it really fits.

That section creates symlinks and sets some udev properties accordingly with the `adb_*` variables. I don't think either name truly reflects that: rule matching is what any `.rules` file does and setting values is more what the new code above is really for. But I don't have an idea for a better name.

Reply:
Seemed best to return it back to "android_usb_rule_match".
It is a little more wordy, but the patch affects less code.
It could be changed again later.